### ISSUE TITLE
.descendantsExcept issue

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#4649]
## steps

https://github.com/lihaoyi/Scalatex

```scala
unmanagedSources in Compile ++= {
  val root = baseDirectory.value.getParentFile
  val exclude = Map(
    "0.13" -> "1.0",
    "1.0" -> "0.13"
  ).apply(sbtBinaryVersion.value)
  val xs = (root / "scalatexSbtPlugin")
    .descendantsExcept("*.scala", s"*sbt-$exclude*")
    .get
  // println(xs.toList)
  xs
}
```

## problem

`.descendantsExcept` doesn't seem to exclude any more, so both 0.13 and 1.0 directories are included.

## expectation

`.descendantsExcept` filters out sbt-0.13 files.

**Comment by eed3si9n** [comment]
Fixed in #4664

**Comment by eed3si9n** [comment]
Tested on M4 to confirm. I need the following change but it seems like it's a plugin bug not ours.

```diff
@@ -13,12 +15,13 @@ object SbtPlugin extends sbt.AutoPlugin {
   val mySeq = Seq(
     scalatexDirectory := sourceDirectory.value / "scalatex",
     managedSources ++= {
-      val inputDir = scalatexDirectory.value
+      val inputDir = scalatexDirectory.value.getAbsoluteFile
```
